src/win32/Com.hxx

```
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 The Music Player Daemon Project
+ * http://www.musicpd.org
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef MPD_WIN32_COM_HXX
+#define MPD_WIN32_COM_HXX
+
+#include <objbase.h>
+#include <windows.h>
+
+class COM {
+public:
+	COM() noexcept : result(CoInitializeEx(nullptr, COINIT_MULTITHREADED)) {}
+	~COM() noexcept {
+		if (*this) {
+			CoUninitialize();
+		}
+	}
+	operator bool() noexcept { return SUCCEEDED(result); }
+	HRESULT Result() const { return result; }
+
+protected:
+	HRESULT result = 0;
```

This initializer is pointless and harmful. Don't initialize variables with arbitrary values - this only hides initialization errors from valgrind.

---

Removed.